selftests/functional/test_test.py

```diff
@@ -0,0 +1,69 @@
+import json
+import os
+import shutil
+import tempfile
+import unittest
+
+from avocado.core import exit_codes
+from avocado.utils import process
+from avocado.utils import script
+
+basedir = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..')
+basedir = os.path.abspath(basedir)
+
+AVOCADO = os.environ.get("UNITTEST_AVOCADO_CMD", "./scripts/avocado")
+
+TEST_SETUP_EXCEPTION = """
+import avocado
+
+class AvocadoTest(avocado.Test):
+
+    def setUp(self):
+        self.log.info('setup code before')
+        raise
```

Comment:
well this works, but is not really a valid way to raise exceptions... (I know, it's just a selftest dummy test, but still people are learning wrong patterns...)